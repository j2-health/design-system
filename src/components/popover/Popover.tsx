import { Popover as AntdPopover, PopoverProps as AntdPopoverProps } from 'antd'
import cx from 'classnames'
import styles from './Popover.module.css'

type _Props = AntdPopoverProps & {
  contentPadding?: boolean
}

export type PopoverProps = Expand<_Props>

const Popover = ({
  children,
  placement = 'top',
  contentPadding = true,
  content,
  ...props
}: PopoverProps) => {
  return (
    <AntdPopover
      overlayClassName={styles.j2Popover}
      {...props}
      placement={placement}
      content={() => {
        if (!contentPadding) {
          return (
            <div className="-m-6">
              {typeof content == 'function' ? content() : content}
            </div>
          )
        }

        return typeof content == 'function' ? content() : content
      }}
    >
      {children}
    </AntdPopover>
  )
}

const PopoverHeader = ({ title }: { title: string }) => {
  return (
    <div
      className={cx(
        'py-2 px-3 items-center flex justify-between',
        styles.popoverHeader
      )}
    >
      {title}
    </div>
  )
}

export { Popover, PopoverHeader }
