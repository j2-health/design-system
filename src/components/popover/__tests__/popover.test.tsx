import { render } from '@testing-library/react'
import { Popover } from '../Popover'

describe('Popover', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Popover title="What is this?" content={<span>The best healthcare</span>}>
        Aetna
      </Popover>
    )
    expect(container).toMatchSnapshot()
  })
})
