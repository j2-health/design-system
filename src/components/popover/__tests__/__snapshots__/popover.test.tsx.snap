// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover should render correctly 1`] = `
<div>
  <div>
    Aetna
  </div>
</div>
`;
